src/app.ts

```diff
@@ -219,6 +237,16 @@ export async function getApp(
     } else {
       app.log.info('Skip registering background jobs, test environment')
     }
+
+    if (isAmqpEnabled) {
+      // This is needed to ensure it gets initialized
+      app.diContainer.cradle.amqpConnectionDisposer
```

Comment:
is this a common pattern in our code? is this an artifact of DI?

Reply:
It's not, it's related to Awilix creator [refusing to implement eager instantiation](https://github.com/jeffijoe/awilix/issues/310). Probably we'll write some kind of a smarter wrapper over Awilix later to fill in the gaps, because we also miss async instantiators, which Awilix also doesn't want to have.